lisa/wlgen/rta.py

```diff
@@ -430,8 +433,45 @@ def _calibrate(cls, target, res_dir):
         if list(pload_list) != sorted(pload_list, reverse=True):
             raise RuntimeError('Calibration values reports big cores less capable than LITTLE cores')
 
+        # Check that the CPU capacities seen by rt-app are similar to the one
+        # the kernel uses
+        true_capacities = cls.get_cpu_capacity_from_calibrations(pload)
+        capa_factors_pct = {
+            cpu: true_capacities[cpu] / cpu_capacities[cpu] * 100
+            for cpu in cpu_capacities.keys()
+        }
+        dispersion_pct = max(abs(100 - factor) for factor in capa_factors_pct.values())
+
+        if dispersion_pct > 2:
+            logger.warning('The calibration values are not inversely proportional to the CPU capacities, the duty cycles will up to {:.2f}% off on some CPUs: {}'.format(dispersion_pct, capa_factors_pct))
```

Comment:
I think you're missing a word in the sentence - also, could we split the line somehow?

Reply:
I can't find any missing word, where do you fail parsing the sentence ?
Splitting lines used to emit user-visible errors is usually a bad idea, for the reason mentioned in the kernel coding style:

> However, never break user-visible strings such as printk messages, because that breaks the ability to grep for them.
https://www.kernel.org/doc/html/v4.10/process/coding-style.html